tests/test_random.py

```diff
@@ -873,3 +873,90 @@ def test_invalid_args(self):
+                                      lambda x: dpnp.asarray(x).astype(dpnp.float32),
+                                      lambda x: dpnp.asarray(x).astype(dpnp.complex64),
+                                      lambda x: dpnp.asarray(x).astype(object),
+                                      lambda x: [(i, i) for i in x],
+                                      lambda x: dpnp.asarray([[i, i] for i in x]),
+                                      lambda x: dpnp.vstack([x, x]).T,
+                                      # gh-11442
+                                      lambda x: (dpnp.asarray([(i, i) for i in x], [
+                                                 ("a", int), ("b", int)]).view(dpnp.recarray)),
+                                      # gh-4270
+                                      lambda x: dpnp.asarray([(i, i) for i in x], [("a", object), ("b", dpnp.int32)])],
+                             ids=[' lambda x: dpnp.array([])',
+                                  ' lambda x: x',
+                                  ' lambda x: dpnp.asarray(x).astype(dpnp.int8)',
+                                  ' lambda x: dpnp.asarray(x).astype(dpnp.float32)',
+                                  ' lambda x: dpnp.asarray(x).astype(dpnp.complex64)',
+                                  ' lambda x: dpnp.asarray(x).astype(object)',
+                                  ' lambda x: [(i, i) for i in x]',
+                                  ' lambda x: dpnp.asarray([[i, i] for i in x])',
+                                  ' lambda x: dpnp.vstack([x, x]).T',
+                                  ' lambda x: (dpnp.asarray([(i, i) for i in x], [("a", int), ("b", int)]).view(dpnp.recarray))',
+                                  ' lambda x: dpnp.asarray([(i, i) for i in x], [("a", object), ("b", dpnp.int32)])]'
+                                  ]
+                             )
+    def test_shuffle1(self, conv):
+        # `conv` contans test lists, arrays (of various dtypes), and multidimensional
+        # versions of both, c-contiguous or not.
+        #
+        # This test is a modification of the original tests of `numpy.random` (both the same):
+        # * tests/test_random.py::TestRandomDist::test_shuffle
+        # * tests/test_randomstate.py::TestRandomDist::test_shuffle
+        #
+        # The original tests do not have a parameterized launch and they
+        # do not correctly check of the results for the dpnp RNG engine.
+
+        # Computing desired 1 dim list for given 1 dim list
+        # on the current device for the given seed number.
+        seed = 1234567890
+
+        dpnp.random.seed(seed)
+        list_1d = [1, 2, 3, 4, 5, 6, 7, 8, 9, 0]
+        dpnp_1d = dpnp.array(list_1d)
+        dpnp_desired_1d = dpnp.random.shuffle(dpnp_1d)
+        desired_1d = [i for i in dpnp_desired_1d]
+        desired_numpy_fallback = [0, 1, 9, 6, 2, 4, 5, 8, 7, 3]
+
+        dpnp.random.seed(seed)
+        alist = conv(list_1d)
```

Comment:
you test diffrent algorithms here by diffrent lambda. Please do not put anything into one test. It is better to use diffrent test cases for diffrent algorithms.

Reply:
Will seperate testing dpnp own implementation and fallback algo on different tests suits.